pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotSegmentUploadDownloadRestletResource.java

```diff
@@ -298,10 +298,14 @@ private SuccessResponse uploadSegment(@Nullable String tableName, TableType tabl
       }
 
       // Zk operations
-      completeZkOperations(enableParallelPushProtection, headers, finalSegmentFile, tableNameWithType, segmentMetadata,
-          segmentName, zkDownloadUri, moveSegmentToFinalLocation, crypterClassName);
-
-      return new SuccessResponse("Successfully uploaded segment: " + segmentName + " of table: " + tableNameWithType);
+      if (completeZkOperations(
+          enableParallelPushProtection, headers, finalSegmentFile, tableNameWithType,
+          segmentMetadata, segmentName, zkDownloadUri, moveSegmentToFinalLocation, crypterClassName)) {
+        return new SuccessResponse("Successfully uploaded segment: " + segmentName + " of table: " + tableNameWithType);
+      } else {
+        return new SuccessResponse("Cannot replace non-existing segment, aborted uploading segment: " + segmentName
```

Comment:
Do we want to return `SucessResponse` in case we failed to replace the segment? Don't we need to return some error response so that the client can handle this accordingly?

Reply:
And the controller metric `CONTROLLER_SEGMENT_UPLOAD_ERROR` might also be required to update as well.

Reply:
Updated to increment the error metric and throw an exception with status code 410 (Gone) instead.